src/jdk.compiler/share/classes/com/sun/tools/javac/jvm/ClassWriter.java

```diff
@@ -862,6 +870,26 @@ int writeRecordAttribute(ClassSymbol csym) {
         return 1;
     }
 
+    int writeParametricAttribute(Symbol sym) {
+        int alenIdx = writeAttr(names.Parametric);
+        int kind = 1; // the default
+        Attribute.Compound c = sym.attribute(syms.parametricType.tsym);
+        Attribute value = c.member(names.kind);
+        if (value != null && value instanceof Attribute.Enum) {
+            Name kindName = ((Attribute.Enum)value).value.name;
+            kind = switch (kindName.toString()) {
+                case "CLASS" -> 1;
+                case "METHOD_ONLY" -> 2;
+                case "METHOD_AND_CLASS" -> 3;
+                default -> throw new AssertionError("unexpected kind");
+            };
+        }
+        databuf.appendChar(poolWriter.putParameter(kind));
```

Comment:
There is something fishy here: note that parameter have an "id" - which can be reused by other entries. That is, you can have 5 different @Parameteric annotations, referring to the same parameter ID. Wouldn't this create duplicate parameters in the constant pool?

As a quick hack - we could pretend that the parameter CP entry had a "name" so that, instead of doing `putParameter`, you do `putParameter(name)` and poolWriter will use existing uniqueness logic to determine if a new parameter has to be emitted.